### ISSUE TITLE
Coupon code based pricing rule not working for multiple pricing rules.

### ISSUE THREAD

**Issue by fadilsiddique** [frappe/erpnext#28601]
## Context information (for bug reports)

**Output of `bench version`**
```
(paste here)
```

## Steps to reproduce the issue

1. Create multiple pricing rule based in coupon code for item groups. Let's say 10%,15%,100%
2. Add items to the cart from point of sale.
3. Apply any coupon code.

### Observed result
Coupon code 15 and 100 percentage discounts 10%. 
### Expected result
coupon with 15% discount should reduce 15% and 100% should reduce 100%.

### Stacktrace / full error message

```
(paste here)
```

## Additional information

OS version / distribution, `ERPNext` install method, etc.

**Comment by fadilsiddique** [comment]
Any updates?

**Comment by fadilsiddique** [comment]
Hi Team,

Any update on this issue?

**Comment by nemesis189** [comment]
@fadilsiddique sorry about the delay, will be looking into it now.

**Comment by nemesis189** [comment]
@fadilsiddique I created 3 pricing rules based on coupon code for the same Item Group. Is this the scenario that you have mentioned?
If so it throws validation for priority of pricing rules:
<img src='https://user-images.githubusercontent.com/36098155/147902795-59bb9291-89ba-4167-8d78-5bbfd03157ff.png' width='700'>

is this the issue?

**Comment by fadilsiddique** [comment]
Yes, this is issue I faced. My case is I've different discount options for same items groups with different coupon codes.

**Comment by fadilsiddique** [comment]
Hi Team,
Any update on this issue?

**Comment by nemesis189** [comment]
@fadilsiddique for this you'll have to assign priority manually
![image](https://user-images.githubusercontent.com/36098155/148721770-e7954622-0c0a-4def-9270-6d1fef49bc7c.png)

**Comment by ahmedadelvo** [comment]
i have the same issue here , any updates ?
